src/ruby/PreProcessor.rb

```diff
@@ -25,7 +26,7 @@ def self.reset
   # Process the emerald to remove indentation and replace with brace convention
   # for an easier time parsing with context free grammar
   def self.process_emerald(file_name)
-    input = File.open(file_name, 'r').read
+    input = File.open(file_name, 'r').read.gsub('{', '\{')
```

Comment:
Nawh, I gotta change it lol. Currently the preprocessor doesn't escape braces, which it should. I'll take it out of this PR right now and make a new PR for it.